### PR TITLE
[xtest] attempt to prevent oom in linking

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Attempt to prevent linkage errors during compilation in cargo x test.

### Have you read the [Contributing Guidelines on pull requests]

y

## Test Plan

ci

## Related PRs

Example failure on a 2xlarge+ machine https://app.circleci.com/pipelines/github/libra/libra/20218/workflows/e3d71fb3-35f6-4bd0-b9ee-d6bb65d3249e/jobs/120474/steps

### CODE REVIEW

File: devtools/x/src/test.rs [hunk 1]
```diff
@@ -76,7 +79,10 @@ pub fn run(mut args: Args, xctx: XContext) -> Result<()> {
     if args.lib {
         direct_args.push(OsString::from("--lib"));
     }
-
+    if args.jobs.is_some() {
```
Comment (bmwill) [inline devtools/x/src/test.rs:82]: Use and “if let” here instead of unwrapping below

Comment (rexhoffman) [reply devtools/x/src/test.rs:82]: @bmwill  Not sure what you're asking here,  

if let jobs = if args.jobs.??? {
}

Comment (rexhoffman) [reply devtools/x/src/test.rs:82]: I'm just going to land for now, so folks aren't blocked.   But I would like to figure out what I should have done here instead.

Comment (rexhoffman) [reply devtools/x/src/test.rs]: pattern matching is a thing, as is Some(var) = Option(somevar) {....  okay interesting.

**Pull request closed without merging**